service/riotapi.py

```diff
@@ -27,6 +30,49 @@ def get_matches(puuid, api_key, num_matches):
     return match_response_list
 
 
+def get_matches_ux(matches, puuid, api_key):
+    matches_dict = {}
+    for match in matches:
+        match_id = match
+        get_match_detail_url = str.format("https://americas.api.riotgames.com/lol/match/v5/matches/{0}?api_key={1}",
+                                          match_id, api_key)
+        match_id_response_dict = get_response_json(get_match_detail_url)
+
+        timestamp = match_id_response_dict.get('info',{}).get('gameCreation') // 1000
+
+        a = datetime.fromtimestamp(timestamp)
+
+        date_time_str = a.strftime("%m-%d %I:%M %p")
+
+        gameDuration = time.strftime('%H:%M:%S', time.gmtime(match_id_response_dict.get('info',{}).get('gameDuration')))
+
+        match_id_dict = {'gameCreation':date_time_str,'gameDuration':gameDuration,'gameMode':match_id_response_dict.get('info',{}).get('gameMode'),'match':match_id}
+
+        participants = match_id_response_dict.get("info").get("participants")
+
+        for participant in participants:
+            if participant.get("puuid") == puuid:
+                match_id_dict['deaths'] = participant.get("deaths")
```

Comment:
This is an area where I am not super certain because of the differences between Python and Java, but it seems like you are building a Model here. Model is a term used to describe when you have a set of data that is purely meant to be kept together and/or visualized. https://www.techtarget.com/whatis/definition/model-view-controller-MVC

In Java, we would create a class that had all these elements and create an instance of the class instead of creating a list and passing it around. Python has classes, so I think we should do the same thing here but I can't be as certain as I would in Java.